src/components/select/select.vue

```diff
@@ -0,0 +1,240 @@
+const sortOrder = ref<string>(props.sortorder)
+
+  defaultTypeMixin(selectSortTypes).verifyType(props.sortorder)
+  ? ''
+  : (sortOrder.value = 'ascending');
+
+
+onMounted(() => {
+  // If sort prop is true, sort list before default option is added to the array
+  if(props.sort) {
+    sortOrder.value == 'ascending'
+      ? optionsList.value.sort((a, b) =>
+          a.text > b.text ? 1 : b.text > a.text ? -1 : 0
+        )
+      : optionsList.value.sort((a, b) =>
+          a.text < b.text ? 1 : b.text < a.text ? -1 : 0
+        )
+
+  }
+
+  const defaultSelect: OptionsProp = {
+    id: '0',
+    value: props.defaultoption ? props.defaultoption : '',
+    text: props.defaultoption ? props.defaultoption : '-- Choose an option --',
+  };
+  optionsList.value.unshift(defaultSelect);
+
+  // If the label position is top - add default option as selected so select container isnt empty
+  labelPosition.value == 'top' ? selectedOption.value = optionsList.value[0].text : null
+});
+
+const popperInstance = computed(() => {
+  const wrapperElem = document.querySelector(`#${selectId.value}`) as HTMLElement;
+  const selectElem = document.querySelector(`.${selectId.value}`) as HTMLElement;
+  
+  return createPopper(wrapperElem, selectElem, {
+    placement: 'auto-start',
+    modifiers: [
+      {
+        name: 'flip',
+        options: {
+          allowedAutoPlacements: [`bottom-end`, `top-end`],
+        },
+      },
+      {
+        name: 'offset',
+        options: {
+          offset: [0, props?.position == 'top' ? 12 : 2],
+        },
+      },
+    ],
+    strategy: 'absolute',
+  });
+});
+
+const handleShowOptions: () => void = () => {
+  showOptions.value = !showOptions.value;
+  if (showOptions.value) {
+    const selectElem = document.getElementById(
+      selectContainerId.value
+    ) as HTMLElement;
+    if (selectElem) {
+      // Only for Vitest
+      selectElem.focus();
+    }
+
+    popperInstance.value.update();
+    selectRef.value.setAttribute('data-show', '');
+  } else {
+    selectRef.value.removeAttribute('data-show');
+  }
+};
+
+const handleEnterShow: () => void = () => {
+  handleShowOptions();
+};
+
+let searchTimer: ReturnType<typeof setTimeout> | null = null;
+const handleSearch: (e: KeyboardEvent) => void = (e) => {
+  if (
+    document.querySelector(`.${selectId.value}`)?.hasAttribute('data-show') &&
+    e.key != 'Enter' &&
+    e.key != 'Alt' &&
+    e.key != 'Tab'
+  ) {
+    if(e.key == 'Space') {
+      searchRef.value = searchRef.value + ' ';
+    } else {
+      searchRef.value = searchRef.value + e.key;
+    }
+  
+    const index = optionsList.value.findIndex(option => option.text.toLowerCase().includes(searchRef.value.toLowerCase()))
+    if(index !== -1) {
+      optionRefs.value[index].focus()
+    }
+
+    if (searchTimer) {
+      return;
+    }
+
+    searchTimer = setTimeout(() => {
+      searchRef.value = '';
+      searchTimer = null;
+    }, 2500);
```

Comment:
I think it's enough with 1s actually,

Reply:
Even if the users are slow they should be able to get 2-3 characters. So is hopefully enough time. 
Changed.